server/src/main/java/org/opensearch/search/aggregations/StarTreeBucketCollector.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.aggregations;
+
+import java.io.IOException;
+
+public abstract class StarTreeBucketCollector extends LeafBucketCollector {
+
+    public abstract void collectStarEntry(int starTreeEntry, long bucket) throws IOException;
```

Comment:
it would be good to have documentation around what's a star tree entry, unless already specified somewhere